File: linguages/python/confluent_kafka_python/consumer.py
from confluent_kafka import Consumer


conf = {
                'bootstrap.servers': 'streaming.eu-frankfurt-1.oci.oraclecloud.com:9092',
                'group.id': 'test1',
                'auto.offset.reset': 'earliest',
                'enable.auto.commit': 'True',
                'security.protocol': 'SASL_SSL',
                'sasl.mechanism': 'PLAIN',
                'sasl.username': '{tenancyName}/{username}/{stream_pool_OCID}',
                'sasl.password': '{authToken}'

            }

consumer = Consumer(conf)
consumer.subscribe(['igorTest'])

While True:
    message = consumer.consume(num_messages=600, timeout=30)
    for msg in message:
        print('Received message : {}{}'.format(msg.value().decode('utf-8'),msg.offset()))

File: languages/java/kafka_confluent/src/main/java/com/oss/kafka/Consumer.java
package com.oss.kafka;

import org.apache.kafka.clients.consumer.ConsumerRecord;
import org.apache.kafka.clients.consumer.ConsumerRecords;
import org.apache.kafka.clients.consumer.KafkaConsumer;

import java.util.*;

public class Consumer {

    private final KafkaConsumer<Integer, String> consumer;
    private final String topic;

    public Consumer(String topic) {
        Properties properties = new Properties();
        properties.put("bootstrap.servers", "streaming.{region}.oci.oraclecloud.com:9092");
        properties.put("security.protocol", "SASL_SSL");
        properties.put("sasl.mechanism", "PLAIN");
        properties.put("sasl.jaas.config", "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"{tenancyName}/{username}/{stream pool OCID}\" password=\"{authToken}\";");
        properties.put("group.id", "test");
        properties.put("enable.auto.commit", "true");
        properties.put("auto.commit.interval.ms", "1000");
        properties.put("session.timeout.ms", "30000");
        properties.put("key.deserializer", "org.apache.kafka.common.serialization.IntegerDeserializer");
        properties.put("value.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");

        consumer = new KafkaConsumer<>(properties);
        this.topic = topic;
    }

    public void testConsumer() {

        consumer.subscribe(Collections.singleton(topic));

        while(true){
            ConsumerRecords<Integer, String> poll = consumer.poll(100);
            for(ConsumerRecord r : poll) {
                System.out.println("key : " + r.key());
                System.out.println(r);
            }
        }

    }

    public static void main(String[] args) {
        String topic = "topic";
        Consumer simpleHLConsumer = new Consumer(topic);
        simpleHLConsumer.testConsumer();
    }

}